feat(metrics): Extract transaction user [INGEST-804] [INGEST-544]

jjbayer opened the pull request:
Extract `user.id` as unique transaction user.

jjbayer committed:
feat(metrics): Extract transaction user

untitaker commented on relay-server/src/metrics_extraction/transactions.rs:
```diff
@@ -318,9 +335,13 @@ mod tests {
 
-        for metric in metrics {
+        for metric in &metrics[0..6] {
             assert!(matches!(metric.value, MetricValue::Distribution(_)));
```
i would keep all the tag-assertions for all metrics except for this one

jjbayer replied:
That makes sense.

jjbayer committed:
Fix changelog, improve test

untitaker committed:
fix(ci): Stop using insecure git protocol for fetching SDK

GitHub deprecated the git protocol for cloning repos recently: https://github.blog/2021-09-01-improving-git-protocol-security-github/

jjbayer committed:
Merge branch 'fix/sentry-sdk-broken-git-ref' into feat/metrics-extract-transaction-user

untitaker committed:
Merge remote-tracking branch 'origin/master' into feat/metrics-extract-transaction-user

jjbayer committed:
test: Attempt to unflake graceful shutdown

jjbayer committed:
Merge remote-tracking branch 'origin/master' into feat/metrics-extract-transaction-user
